File: src/main/GO/leetcode/lists/offer/Offer27.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/29 19:12
 */

func mirrorTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}
	root.Right, root.Left = root.Left, root.Right
	mirrorTree(root.Left)
	mirrorTree(root.Right)
	return root
}

File: src/main/java/leetcode/lists/offer/Offer27.java
package leetcode.lists.offer;

import common.TreeNode;

/**
 * @author jingxinwu
 * @date 2021-11-21 6:37 下午
 */
public class Offer27 {

    //镜像树
    public TreeNode mirrorTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode temp = root.left;
        root.left = root.right;
        root.right = temp;

        mirrorTree(root.left);
        mirrorTree(root.right);
        return root;

    }
}
